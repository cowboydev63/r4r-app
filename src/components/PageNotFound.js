import React from 'react';
import PropTypes from 'prop-types';
import Error from './Error'
import { connect } from 'react-redux';
import { pageNotFound } from '../state/error/actions';

export class PageNotFound extends React.Component {
    static propTypes = {
        pageNotFound: PropTypes.func.isRequired
    }
    
    componentDidMount(){
        this.props.pageNotFound()
    }

    render(){
        return(
            <Error title="Page not found" body="We can't find the page you're looking for." showRedirect={true} />
        )
    }
}

export default connect(null, { pageNotFound })(PageNotFound);