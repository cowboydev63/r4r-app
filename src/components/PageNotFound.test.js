import React from 'react';
import { shallow } from 'enzyme';
import { PageNotFound } from './PageNotFound';

describe('PageNotFound Component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<PageNotFound pageNotFound={()=> {}}/>);
        expect(wrapper).toMatchSnapshot();
    })

    it('calls componentDidMount once', () => {
        const componentDidMountSpy = jest.spyOn(PageNotFound.prototype, 'componentDidMount');

        const wrapper = shallow(<PageNotFound pageNotFound={()=> {}}/>);

        expect(PageNotFound.prototype.componentDidMount).toHaveBeenCalledTimes(1);

        componentDidMountSpy.mockClear();
    })
    it('dispatches an action on load indicating 404 load', () => {
        const mockFn = jest.fn()
        const wrapper = shallow(<PageNotFound pageNotFound={mockFn}/>);
        expect(mockFn).toHaveBeenCalledTimes(1);
    })
})