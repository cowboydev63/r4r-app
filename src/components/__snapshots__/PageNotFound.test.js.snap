// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound Component renders correctly 1`] = `
<Error
  body="We can't find the page you're looking for."
  showRedirect={true}
  title="Page not found"
/>
`;
